### ISSUE TITLE
Promise does not resolve is error occurs in the middle of the handler chain

### ISSUE THREAD

**Issue by pluby** [hoangvvo/next-connect#164]
I had an error handling issue and wrote this test for it but was unable to resolve the problem:

```
  it("reject if there is an callback error with penultimate middleware", () => {
    const handler = nc();
    handler.use((req, res, next) => {
      next(Error("error :("))
    }, (req, res, next) => {
      res.end("good")
    });
    const app = createServer(async (req, res) => {
      try {
        await handler.run(req, res);
        res.end("good");
      } catch (e) {
        // Don't use same text as the error otherwise the default error
        // handler may cause the test to pass.
        res.end("bad");
      }
    });
    return request(app).get("/").expect("bad");
  });
```

There are two issues revealed by this test:

* The default error handler returns the error message as the response, when it should cause the promise to reject triggering the `res.end("bad")` code.
* The promise is never settled so Next never gets to see there error and there is no way to trap it.

**Comment by hoangvvo** [comment]
Sorry for the issue and the late response, this seems to be a misbehavior of the current logic. To fix it temporary, in `onError`, simply call `next(err)`. I will fix it this week. So sorry again!

**Comment by hoangvvo** [comment]
This should be fixed in ad857bedb0996312ad3a5ea966ce3a60417429a6. Please reopen if it has not. Thank you.

**Comment by pluby** [comment]
Thanks.
